sympy/solvers/ode/tests/test_ode.py

```diff
@@ -1446,124 +1446,6 @@ def test_1st_exact2_broken_check():
     assert checkodesol(eq, sol, order=1, solve_for_func=False)[0]
 
 
-def test_separable1():
-    # test_separable1-5 are from Ordinary Differential Equations, Tenenbaum and
-    # Pollard, pg. 55
-    eq1 = f(x).diff(x) - f(x)
-    eq2 = x*f(x).diff(x) - f(x)
-    eq3 = f(x).diff(x) + sin(x)
-    eq4 = f(x)**2 + 1 - (x**2 + 1)*f(x).diff(x)
-    eq5 = f(x).diff(x)/tan(x) - f(x) - 2
-    eq6 = f(x).diff(x) * (1 - sin(f(x))) - 1
-    sol1 = Eq(f(x), C1*exp(x))
-    sol2 = Eq(f(x), C1*x)
-    sol3 = Eq(f(x), C1 + cos(x))
-    sol4 = Eq(f(x), tan(C1 + atan(x)))
-    sol5 = Eq(f(x), C1/cos(x) - 2)
-    sol6 = Eq(-x + f(x) + cos(f(x)), C1)
-    assert dsolve(eq1, hint='separable') == sol1
-    assert dsolve(eq2, hint='separable') == sol2
-    assert dsolve(eq3, hint='separable') == sol3
-    assert dsolve(eq4, hint='separable') == sol4
-    assert dsolve(eq5, hint='separable') == sol5
-    assert dsolve(eq6, hint='separable') == sol6
-    assert checkodesol(eq1, sol1, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq2, sol2, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq3, sol3, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq4, sol4, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq5, sol5, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq6, sol6, order=1, solve_for_func=False)[0]
-
-
-@slow
-def test_separable2():
-    a = Symbol('a')
-    eq6 = f(x)*x**2*f(x).diff(x) - f(x)**3 - 2*x**2*f(x).diff(x)
-    eq7 = f(x)**2 - 1 - (2*f(x) + x*f(x))*f(x).diff(x)
-    eq8 = x*log(x)*f(x).diff(x) + sqrt(1 + f(x)**2)
-    eq9 = exp(x + 1)*tan(f(x)) + cos(f(x))*f(x).diff(x)
-    eq10 = (x*cos(f(x)) + x**2*sin(f(x))*f(x).diff(x) -
-            a**2*sin(f(x))*f(x).diff(x))
-    sol6 = Eq(Integral((u - 2)/u**3, (u, f(x))),
-        C1 + Integral(x**(-2), x))
-    sol7 = Eq(-log(-1 + f(x)**2)/2, C1 - log(2 + x))
-    sol8 = Eq(asinh(f(x)), C1 - log(log(x)))
-    # integrate cannot handle the integral on the lhs (cos/tan)
-    sol9 = Eq(Integral(cos(u)/tan(u), (u, f(x))),
-        C1 + Integral(-exp(1)*exp(x), x))
-    sol10 = Eq(-log(cos(f(x))), C1 - log(- a**2 + x**2)/2)
-    assert dsolve(eq6, hint='separable_Integral').dummy_eq(sol6)
-    assert dsolve(eq7, hint='separable', simplify=False) == sol7
-    assert dsolve(eq8, hint='separable', simplify=False) == sol8
```

Comment:
```
>>> eq=x*log(x)*f(x).diff(x) + sqrt(1 + f(x)**2)
>>> dsolve(eq,hint='separable',simplify=False)

Eq(asinh(f(x)), C1 - log(log(x)))

>>> checkodesol(eq,_)
(False, -2*sqrt(sinh(C1 - log(log(x)))**2 + 1)*exp(C1)*log(x) + exp(2*C1) + log(x)**2)
```
@oscarbenjamin example 8 gives incorrect solution, i don't know how it passes on master as it also checks chceckodesol same way

Reply:
It looks like checkodesol is unable to simplify the residual in this case. The solution is correct but perhaps not for all complex x. We have:
```julia
In [88]: eq = x*log(x)*f(x).diff(x) + sqrt(1 + f(x)**2)                                                                           

In [89]: eq                                                                                                                       
Out[89]: 
                       ___________
         d            ╱  2        
x⋅log(x)⋅──(f(x)) + ╲╱  f (x) + 1 
         dx                       

In [90]: sol = dsolve(eq)                                                                                                         

In [91]: sol                                                                                                                      
Out[91]: f(x) = sinh(C₁ - log(log(x)))

In [92]: eq.subs(sol.lhs, sol.rhs)                                                                                                
Out[92]: 
                                         _____________________________
         ∂                              ╱     2                       
x⋅log(x)⋅──(sinh(C₁ - log(log(x)))) + ╲╱  sinh (C₁ - log(log(x))) + 1 
         ∂x                                                           

In [93]: eq.subs(sol.lhs, sol.rhs).doit()                                                                                         
Out[93]: 
   _____________________________                         
  ╱     2                                                
╲╱  sinh (C₁ - log(log(x))) + 1  - cosh(C₁ - log(log(x)))

In [94]: eq.subs(sol.lhs, sol.rhs).doit().simplify()                                                                              
Out[94]: 
   _____________________________                         
  ╱     2                                                
╲╱  sinh (C₁ - log(log(x))) + 1  - cosh(C₁ - log(log(x)))
```
What we have here is something like this:
```julia
In [95]: res = sqrt(sinh(x)**2 + 1) - cosh(x)                                                                                     

In [96]: res                                                                                                                      
Out[96]: 
   ______________          
  ╱     2                  
╲╱  sinh (x) + 1  - cosh(x)

In [97]: simplify(res)                                                                                                            
Out[97]: 
   __________          
  ╱     2              
╲╱  cosh (x)  - cosh(x)
```
The sqrt does not evaluate because `cosh(x)` is not known to be positive. It is positive if `x` is real but that isn't given in the assumptions:
```julia
In [99]: res.subs(x, Symbol('x', real=True)).simplify()                                                                           
Out[99]: 0
```
We need to know that C1 and x are real in order to simplify the result.

Reply:
The given solution might only be valid for x>1.

There is another solution to the ODE given by `f(x) = I`:
```julia
In [110]: eq.subs(f(x), I).doit()                                                                                                 
Out[110]: 0
```

Reply:
@oscarbenjamin if we declare C1 as` real=True`, then other examples will fail.

Reply:
I think we just have to not call use checkodesol for this example.

Reply:
so should i mark this as xfail for now?

Reply:
We should still test dsolve but not checkodesol. Maybe there needs to be a way to flag that in the tests.

Reply:
> We should still test dsolve but not checkodesol. Maybe there needs to be a way to flag that in the tests.

so should i add a flag as key in example like `don't_checkodesol = True`?

Reply:
Yes or perhaps `'checkodesol': 'XFAIL'` or something.